saige_assoc.py

```diff
@@ -0,0 +1,383 @@
+    This will run a single variant test using a score test
+
+    Input:
+    - vcfFile: path to VCF file
+    - vcfFileIndex: path to VCF index file (tbi)
+    - saige output path: path to output saige file
+    - chrom: chromosome to run this on
+    - cis window: file with chrom | start | end to specify window
+    - GMMAT model file: null model fit from previous step (.rda)
+    - Variance Ratio file: as estimated from previous step (.txt)
+    - group annotation: select only some annos from group file (e.g., lof)
+    - group file: for each gene/set,
+        one row specifying variants,
+        one row specifying each variant's anno,
+        one optional row with all weights
+    - allele order: specifying whether alt-first or ref-first in genotype files
+    - min MAF: minimum variant minor allele frequency to include
+    - min MAC: minimum variant minor allele count to include
+    - LOCO: leave one chromosome out (for what specifically?)
+    - is no adjusted cov: covariate adjustment?
+    - specify whether we're using a sparse GRM (vs full? vs no GRM at all?)
+
+    Output:
+    Rscript command (str) ready to run
+    """
+    saige_command_step2_sv = 'Rscript step2_tests_qtl.R'
+    saige_command_step2_sv += f' --vcfFile={vcf_file}'
+    saige_command_step2_sv += f' --vcfFileIndex={vcf_file_index}'
+    saige_command_step2_sv += f' --SAIGEOutputFile={saige_output_file}'
+    saige_command_step2_sv += f' --chrom={chrom}'
+    saige_command_step2_sv += f' --minMAF={min_maf}'
+    saige_command_step2_sv += f' --minMAC={min_mac}'
+    saige_command_step2_sv += f' --LOCO={loco_bool}'
+    saige_command_step2_sv += f' --GMMATmodelFile={gmmat_model_path}'
+    saige_command_step2_sv += f' --SPAcutoff={spa_cutoff}'
+    saige_command_step2_sv += f' --varianceRatioFile={variance_ratio_path}'
+    # saige_command_step2_sv += f' --is_fastTest={is_fast_test}'
+    saige_command_step2_sv += f' --rangestoIncludeFile={cis_window_file}'
+    saige_command_step2_sv += f' --markers_per_chunk={n_markers}'
+    # saige_command_step2_sv += f' --pval_cutoff_for_fastTest={pval_cutoff}'
+    return saige_command_step2_sv
+
+
+def build_obtain_gene_level_pvals_command(
+    gene_name: str,
+    saige_sv_output_file: str,
+    saige_gene_pval_output_file: str,
+):
+    """Build SAIGE command to obtain gene-level pvals
+    Only for single-variant tests (Step3)
+    combines single-variant p-values to obtain one gene
+    level p-valye
+
+    Input:
+    - ouput of previous step, association file (txt)
+    - gene we need to aggregate results for (across SNPs)
+    - path for output file
+    """
+    saige_command_step3 = 'Rscript step3_gene_pvalue_qtl.R'
+    saige_command_step3 += f' --assocFile={saige_sv_output_file}'
+    saige_command_step3 += f' --geneName={gene_name}'
+    saige_command_step3 += f' --genePval_outputFile={saige_gene_pval_output_file}'
+    return saige_command_step3
+
+
+# # Run gene set association (for RVs)
```

Comment:
is this in or out?

Reply:
sorry what do you mean by this?

Reply:
This whole section is commented out, along with a section further down. Are these going to be included in this PR, or can they be removed? (this PR is still in draft, so maybe these are being worked on)

Reply:
i see sorry you mean from line 175 down! I thought you were referring to the return line from the previous function.

I'll work on this section now, I think it makes sense to include it in this PR, to test all functions from the package with test data.

the section below on the other hand i left in because it is how in practice we'll run this (looping over cell types, chromosomes, genes), whereas the test data only has one gene to test so no looping is needed. I guess that should be a separate PR then? Sorry I feel like i need to test that the package works first, then set up the pipeline with real data but maybe this is not the way to go about this